torchdynamo/eval_frame.py

```diff
@@ -396,7 +397,7 @@ def guard_export_print(guards):
     return explanation, out_guards, graphs, ops_per_graph
 
 
-def export(f, *args, **kwargs):
+def export(f, *args, aten_graph=False, **kwargs):
```

Comment:
We might need to expose decomp_table and use_functionalization in the export API latter.

Reply:
Agreed.

Reply:
seems like those functionalities are better exposed as passes?

Reply:
Functionalization actually requires running the code and retracing, so it is a bit more involved than just "running a pass" and could benefit from already having the input args available here for a seamless user experience.

This is very timely -- I was just playing with this locally for Torch-MLIR and I hit a few stumbling blocks with running functionalization and scripting the result.

Reply:
Glad it could be of use! There's more I plan to add here, happy to write something up.

Reply:
One feature request would be to support dynamic shapes. For example, to be able to say something like `torchdynamo.export(f, [TensorPlaceholder([-1, 12])])` -- we have an analogous API in Torch-MLIR [link](https://github.com/llvm/torch-mlir/blob/e6528f701a7cba8c0e10195bc4b838c143798fa0/python/torch_mlir/__init__.py#L74) and we would like to be able to use TorchDynamo under the hood rather than torch.jit.script/torch.jit.trace which each have their own limitations (script: obvious, trace: limited to static shapes).

Reply:
That's a good suggestion. We have symbolic shape support already, but not as an input placeholder. Let me mull this over and check out your API and see what we can do here.

Reply:
You can see examples of the API here: https://github.com/llvm/torch-mlir/blob/e6528f701a7cba8c0e10195bc4b838c143798fa0/python/test/compile_api/basic.py#L1

Reply:
Took a really rough stab at it here - needs a lot more tests and edge cases handled, but probably a good conversation starting PR to kind of see which direction we want to take it in https://github.com/pytorch/torchdynamo/pull/1123/